resolver-dns/src/main/java/io/netty/resolver/dns/DefaultDnsCache.java

```diff
@@ -301,21 +317,53 @@ void add(DefaultDnsCacheEntry e) {
                                 assert replacedEntry == null;
                                 replacedEntry = entry;
                             }
-                        }
+                            i++;
+                        } while (i < entries.size());
                         newEntries.add(e);
-                        if (compareAndSet(entries, newEntries)) {
+                        if (compareAndSet(entries, Collections.unmodifiableList(newEntries))) {
                             if (replacedEntry != null) {
```

Comment:
this is an `if` block with no code contents. consider removing the if block and just keep the comments.